HIP/hip-850.md

```diff
@@ -0,0 +1,69 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Wallet
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Review
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks
+updated: 2024-01-24
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury wallet. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
+
+## Motivation
+
+Currently, there is a limitation in dynamically updating NFTs in a controlled manner. The proposed enhancement to the Supply Key functionality addresses this gap, allowing for secure and selective updates of NFTs within the treasury wallet, a feature particularly relevant for various evolving use cases in the Hedera ecosystem.
+
+## Rationale
+
+The enhancement focuses on granting the Supply Key the capability to update the data of an NFT serial number exclusively while it is in the treasury wallet. This approach ensures security and control, preventing unauthorized modifications once the NFT is distributed.
```

Comment:
Q: does this imply the modification may only be done prior to first distribution?
What if the treasury is sent the NFT after it previously distributed it, can it modify it then?
What about an NFT that was minted straight to the user and didn't go to the treasury first, if it is sent to the treasury can the treasury modify it?

Reply:
I think this HIP needs to be reconciled with [HIP-657: Mutable metadata fields for dynamic NFTs](https://hips.hedera.com/hip/hip-657), which introduces the collection level `metadata_key`, which can be used to modify the currently immutable individual NFT metadata via the new `TokenUpdateNftsTransactionBody`.

I could be mistaken, but my read is that as long as the `metadata_key` is defined for the token/collection, NFT metadata can be manipulated at any time, regardless of the of the current NFT owner. There are no notions of a treasury wallet, post-distribution or metadata key interactions with the supply key.

Reply:
@Nana-EC 
_does this imply the modification may only be done prior to first distribution?_
No, this be done at any time when the token is in the treasury wallet, just like how burning tokens with the supply key currently works

_What if the treasury is sent the NFT after it previously distributed it, can it modify it then?_
Yes, anytime the token is in the treasury wallet, the supply key can modify the metadata field. 

What about an NFT that was minted straight to the user and didn't go to the treasury first, if it is sent to the treasury can the treasury modify it?
Yes, matching parity with how token burning works currently, this supply key adapting metadata of token will work the same exact way.

Reply:
@jascks 
For the history of this HIP, it was pulled out of 657 into it's own to make the HIPs smaller and easier to get implemented. The functionality here is separate. 

This would create a dynamic that does not exisit on any network that I know of, where a NFT would become semi-fungible. The metadata could be updated ONLY when in the treasury. 

So if you had a game piece, and you need it updated with new stats earned in a game, if the NFT is sent to the treasury, it can be updated without destroying the provenance. 

When it's sent back to the user, the creator can't manipulate it which provides a layer of security and opens the door to some innovative dynamics that don't exist anywhere else.

Reply:
Interesting. The concept that a disbursed NFT could have it's metadata changed upon return to the treasury has me thinking.
I guess the serial id doesn't change so it remain identifiable but i'm thinking of real world scenarios where the side effect may not be clear.
Thanks for clarifying